File: base.database/src/main/java/alvin/database/domain/models/Person.java
package alvin.database.domain.models;

import java.time.LocalDate;

public class Person {
    private final int id;
    private final String name;
    private final Gender gender;
    private final LocalDate birthday;

    public Person(int id, String name, Gender gender, LocalDate birthday) {
        this.id = id;
        this.name = name;
        this.gender = gender;
        this.birthday = birthday;
    }

    public Person(String name, Gender gender, LocalDate birthday) {
        this.id = 0;
        this.name = name;
        this.gender = gender;
        this.birthday = birthday;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Gender getGender() {
        return gender;
    }

    public LocalDate getBirthday() {
        return birthday;
    }
}

File: adv.permission/src/main/java/alvin/adv/permission/storage/models/Person.kt
package alvin.adv.permission.storage.models

import alvin.lib.mvp.contracts.IModel
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.time.LocalDate

class Person
@JsonCreator constructor(
        @JsonProperty("name") val name: String,
        @JsonProperty("gender") val gender: Gender,
        @JsonProperty("birthday") val birthday: LocalDate,
        @JsonProperty("remark") val remark: String
) : IModel
